### PR TITLE
[DO NOT MERGE UNTIL THE FEATURE IS RELEASED] MP-5722: Merchant switcher

### PR DESCRIPTION
**Pull request by zssamoylov**
- Ticket: https://spryker.atlassian.net/browse/MP-5722

| MODULE     | DESCRIPTION                |
|------------|----------------------------|
| MerchantSwitcher | Provides functionality for switching merchant in a quote and in quote items   |
| MerchantSwitcherWidget | Allows to select a merchant from an active merchants list and store the selected merchant data|
| ShopContextExtension | Provides an interfaces of plugins to extend ShopContext module from the other modules   |
| WishlistExtension | Provides plugin interfaces for extending Wishlist module functionality   |
| CartExtension | Provides plugin interfaces used by Cart module   |
| CheckoutExtension | Provides plugin interfaces used by Checkout module   |

### CODE REVIEW

File: docs/marketplace/dev/feature-walkthroughs/202108.0/merchant-switcher-feature-walkthrough.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+|------------|----------------------------|
+| MerchantSearch | Manages Elasticsearch documents for merchant entities   |
+| MerchantSwitcher | Provides functionality for switching merchant in a quote and in quote items   |
+| MerchantSwitcherWidget | Allows to select a merchant from an active merchants list and store the selected merchant data   |
+
+## Related Developer articles
+
+| INTEGRATION GUIDES | GLUE API GUIDES  |
+| ------------- | -------------- |
+| [Marketplace Wishlist feature integration](/docs/marketplace/dev/feature-integration-guides/{{page.version}}/merchant-switcher-feature-integration.html) | [Managing wishlists](/docs/marketplace/dev/glue-api-guides/{{page.version}}/wishlists/managing-wishlists.html)
```
Comment (ilyakubanov) [inline docs/marketplace/dev/feature-walkthroughs/202108.0/merchant-switcher-feature-walkthrough.md:26]: why wishlist?

File: docs/marketplace/dev/feature-walkthroughs/202108.0/merchant-switcher-feature-walkthrough.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+title: Merchant Switcher feature walkthrough
+last_updated: Sep 29, 2021
+description: This article provides technical details on the Merchant Switcher feature.
+template: merchant-switcher-feature-walkthrough
+---
+
+With the *Merchant Switcher* feature, the customers can select an active merchant so that he can only see products and offers which belong to this merchant.
+Has been created plugins to check that chosen merchant is equal in all items in Cart, Quote, Checkout modules.
+So if merchant is not equal the item replaces with offer or merchant product from owned to chosen merchant.
+If  the system doesn't have suitable items, a customer will see the error or warning about these items.
```
Comment (ilyakubanov) [inline docs/marketplace/dev/feature-walkthroughs/202108.0/merchant-switcher-feature-walkthrough.md:11]:
```suggestion
The feature modules provide plugins to validate merchant in a quote on cart and checkout as well as in wishlist items.
So if merchant is not equal - finds substitution for items depending on the chosen merchant.
If  the system doesn't have suitable items, a customer will see the error or warning about these items.
```
As I understand it also works with wishlist, right?

File: docs/marketplace/dev/feature-walkthroughs/202108.0/merchant-switcher-feature-walkthrough.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+## Module dependency graph
+![Entity Diagram](https://confluence-connect.gliffy.net/embed/image/8db03d24-88d4-4715-a5e1-afae4f2ff8ca.png?utm_medium=live&utm_source=confluence)
+
+| MODULE     | DESCRIPTION                |
+|------------|----------------------------|
+| MerchantSearch | Manages Elasticsearch documents for merchant entities   |
+| MerchantSwitcher | Provides functionality for switching merchant in a quote and in quote items   |
```
Comment (ilyakubanov) [inline docs/marketplace/dev/feature-walkthroughs/202108.0/merchant-switcher-feature-walkthrough.md:19]: please mention a wishlist in description

File: docs/marketplace/dev/feature-walkthroughs/202108.0/merchant-switcher-feature-walkthrough.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+---
+
+With the *Merchant Switcher* feature, the customers can select an active merchant so that he can only see products and offers which belong to this merchant.
+Has been created plugins to check that chosen merchant is equal in all items in Cart, Quote, Checkout modules.
+So if merchant is not equal the item replaces with offer or merchant product from owned to chosen merchant.
+If  the system doesn't have suitable items, a customer will see the error or warning about these items.
+
+## Module dependency graph
+![Entity Diagram](https://confluence-connect.gliffy.net/embed/image/8db03d24-88d4-4715-a5e1-afae4f2ff8ca.png?utm_medium=live&utm_source=confluence)
+
```
Comment (ilyakubanov) [inline docs/marketplace/dev/feature-walkthroughs/202108.0/merchant-switcher-feature-walkthrough.md:15]: - MerchantSearch  doesn't belong to this feature
- MerchantSwitcher doesn't depend on Checkout
- MerchantSwitcher provides functionality by introducing some plugins for Checkout, Cart, Wishlist so maybe we need to add extension modules to diagram instead for better context understanding?
- please add an optional module

Comment (zssamoylov) [reply docs/marketplace/dev/feature-walkthroughs/202108.0/merchant-switcher-feature-walkthrough.md:15]: updated.

File: docs/marketplace/dev/feature-walkthroughs/202108.0/merchant-switcher-feature-walkthrough.md (hunk 3 above)
Comment (zssamoylov) [reply docs/marketplace/dev/feature-walkthroughs/202108.0/merchant-switcher-feature-walkthrough.md:18]: fixed.

**Review by ilyakubanov — DISMISSED** [review]

**Pull request closed without merging**